src/commands/flow/scan.ts

```diff
@@ -24,66 +34,65 @@ export default class scan extends SfdxCommand {
 
   protected static requiresUsername = false;
   protected static supportsDevhubUsername = false;
-  protected static requiresProject = false;
+  public static requiresProject = false;
   protected static supportsUsername = true;
 
   protected userConfig;
   protected failOn = "error";
   protected errorCounters: Map<string, number> = new Map<string, number>();
 
-  protected static flagsConfig = {
-    directory: flags.filepath({
+  public static readonly flags = {
+    directory: Flags.directory({
       char: "d",
       description: messages.getMessage("directoryToScan"),
       required: false,
+      exists: true,
     }),
-    config: flags.filepath({
+    config: Flags.file({
       char: "c",
       description: "Path to configuration file",
       required: false,
     }),
-    failon: flags.enum({
+    failon: Flags.option({
       char: "f",
-      description: "Thresold failure level (error, warning, note, or never) defining when the command return code will be 1",
-      options: ["error", "warning", "note", "never"],
+      description: "Threshold failure level (error, warning, note, or never) defining when the command return code will be 1",
+      options: ["error", "warning", "note", "never"] as const,
       default: "error"
-    }),
-    retrieve: flags.boolean({
+    })(),
+    retrieve: Flags.boolean({
       char: "r",
       description: "Force retrieve Flows from org at the start of the command",
       default: false,
     }),
-    sourcepath: flags.filepath({
+    sourcepath: Flags.directory({
       char: "p",
-      description: "comma-separated list of source flow paths to scan",
+      description: "Comma-separated list of source flow paths to scan",
       required: false,
     }),
+    targetusername: Flags.string({
+      char: 'u',
+      description: 'Retrieve the latest metadata from the target before the scan.',
+      required: false,
+    })
```

Comment:
Another thing to note: I'm not sure what happened to the `targetusername` flag? I think an older version of `@salesforce/core` had this as a default flag, but now we need to explicitly create it.

Reply:
Hmmm gotcha. I haven't used this with flow scanner due to my usage contained in the scanning directory. All invocations of sf cli core gets the default username property to set the target org

Reply:
But yea you're absolutely right, scan.ts has defaultusername which calls retrive on org